agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFile.java

```diff
@@ -109,6 +112,14 @@
 
     logDir = initLogDir();
     directory = logDir + STATUS_FILE_DIRECTORY;
+    shouldWrite =
+        DiagnosticsHelper.useAppSvcRpIntegrationLogging() && new File(directory).canWrite();
+    if (!shouldWrite) {
+      LoggerFactory.getLogger(StatusFile.class)
+          .info(
+              "Detected running on a read-only file system. Status json file won't be created. If this is unexpected, please check that process has write access to the directory: {}",
+              directory);
+    }
```

Comment:
oh, I think this is too early to log. let's move this to `MainEntryPoint.start()` right after `startupLogger` is configured.

also, I think the condition here needs to be `DiagnosticsHelper.useAppSvcRpIntegrationLogging() && !shouldWrite`, otherwise it will be logged outside of app services also

Reply:
shouldWrite includes `DiagnosticsHelper.useAppSvcRpIntegrationLogging()` this check though.

Reply:
the condition is `!shouldWrite`, so I believe it will include standalone deployments

Reply:
MainEntryPoint.start invoked `StatusFile.putValueAndWrite`, which has the shouldWrite check.. i think it's redundant checking twice.  putValueAndWrite is also used elsewhere.  would it be more cleaner to check in all the callers?

before invoking StatusFile methods, check shouldWrite first. Now, i get your point. it should check `DiagnosticsHelper.useAppSvcRpIntegrationLogging()` again when `shouldWrite` is false to log only when it's app service

Reply:
another option is to pass startupLogger to StatusFile.

Reply:
> would it be more cleaner to check in all the callers?

this sounds good, probably just need an AtomicBoolean to only log it once 👍

Reply:
oh right, I got confused, this is tricky because of the delayed logging init. ya, I kind of like that idea to pass the startupLogger in to StatusFile. if it's null we know it's too early

Reply:
i changed it to check shouldWrite in all callers.